tests/suites/test_suite_ssl.function

```diff
@@ -2976,3 +2968,148 @@ exit:
+    TEST_ASSERT( ret == 0 );
+
+    ret = mbedtls_move_handshake_to_state( &(client.ssl),
+                                           &(server.ssl),
+                                           MBEDTLS_SSL_HANDSHAKE_OVER );
+    TEST_ASSERT( ret == 0 );
+    TEST_ASSERT( client.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
+    TEST_ASSERT( server.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
+
+    /* Perform this test with two message types. At first use a message
+     * consistiong only 0x00 for the client and only 0xFF for the server.
+     * At the second time use message with generated data */
+    for( int msg_type = 0; msg_type < 2; msg_type++ )
+    {
+        int cli_writen = 0;
+        int srv_writen = 0;
+        int cli_read = 0;
+        int srv_read = 0;
+        int cli_fragments = 0;
+        int srv_fragments = 0;
+
+        if( msg_type == 0 )
+        {
+            memset( cli_msg_buf, 0x00, cli_msg_len );
+            memset( srv_msg_buf, 0xff, srv_msg_len );
+        }
+        else
+        {
+            int j = 0;
+            for( int i = 0; i < cli_msg_len; i++ )
+                cli_msg_buf[i] = j++ & 0xFF;
+            for( int i = 0; i < srv_msg_len; i++ )
+                srv_msg_buf[i] = ( j -= 5 ) & 0xFF;
+        }
+
+        while( cli_read < srv_msg_len || srv_read < cli_msg_len )
+        {
+            /* Client sending */
+            if( cli_msg_len > cli_writen )
+            {
+                ret = mbedtls_ssl_write( &(client.ssl), cli_msg_buf + cli_writen, cli_msg_len - cli_writen );
+                if( ret >= 0 )
+                {
+                    TEST_ASSERT( ret <= cli_msg_len );
+                    cli_fragments++;
+                    cli_writen += ret;
+                }
+                else
+                {
+                    TEST_ASSERT( ret == MBEDTLS_ERR_SSL_WANT_WRITE );
+                }
+            }
+
+            /* Server sending */
+            if( srv_msg_len > srv_writen )
+            {
+                ret = mbedtls_ssl_write( &(server.ssl), srv_msg_buf + srv_writen, srv_msg_len - srv_writen );
+                if( ret >= 0 )
+                {
+                    TEST_ASSERT( ret <= srv_msg_len );
+                    srv_writen += ret;
+                    srv_fragments++;
+                }
+                else
+                {
+                    TEST_ASSERT( ret == MBEDTLS_ERR_SSL_WANT_WRITE );
+                }
+            }
+
+            /* clients reading */
+            if( cli_read < srv_msg_len )
+            {
+                ret = mbedtls_ssl_read( &(client.ssl), cli_in_buf + cli_read, srv_msg_len - cli_read );
+                if( ret >= 0 )
+                {
+                    TEST_ASSERT( ret <= srv_msg_len );
+                    cli_read += ret;
+                }
+                else
+                {
+                    TEST_ASSERT( ret == MBEDTLS_ERR_SSL_WANT_READ );
+                }
+            }
+
+            /* Server reading */
```

Comment:
The code for "Client reading" and "Server reading" are very similar. Would you be able to refactor this into a shared function (https://refactoring.guru/smells/duplicate-code)?

Reply:
as above